secure/storage/vault/src/lib.rs

```diff
@@ -217,20 +238,32 @@ impl Client {
     }
 
     pub fn create_ed25519_key(&self, name: &str, exportable: bool) -> Result<(), Error> {
-        let request = ureq::post(&format!("{}/v1/transit/keys/{}", self.host, name));
+        let request = self
+            .agent
+            .post(&format!("{}/v1/transit/keys/{}", self.host, name));
         let resp = self
             .upgrade_request(request)
             .send_json(json!({ "type": "ed25519", "exportable": exportable }));
         match resp.status() {
-            200 => Ok(()),
-            204 => Ok(()),
+            200 => {
+                // Explicitly clear buffer so the stream can be re-used.
+                resp.into_string()?;
+                Ok(())
+            }
+            204 => {
+                // Explicitly clear buffer so the stream can be re-used.
+                resp.into_string()?;
+                Ok(())
+            }
             404 => Err(Error::NotFound("transit/".into(), name.into())),
             _ => Err(resp.into()),
         }
     }
 
     pub fn delete_key(&self, name: &str) -> Result<(), Error> {
-        let request = ureq::post(&format!("{}/v1/transit/keys/{}/config", self.host, name));
+        let request = self
+            .agent
+            .post(&format!("{}/v1/transit/keys/{}/config", self.host, name));
```

Comment:
Sorry for the bad line, but does 271 also need to clear the buffer if it's okay?

Reply:
yes... so hacky... but yes

Reply:
it's probably better to have a wrapper around and make it less hacky